internal/executor/executor.go

```diff
@@ -36,6 +40,9 @@ func New(projectDir string, tasks []*api.Task, opts ...Option) (*Executor, error
 		e.logger.Out = ioutil.Discard
 	}
 
+	// Filter tasks (e.g. if a user wants to run only a specific task without dependencies)
+	tasks = e.taskFilter(tasks)
```

Comment:
I'm afraid it will break `GetNextTask` which can now return `nil` in this case when filtering for `bar` task:

```yaml
foo_task:
  script: echo 'Foo'

bar_task:
  depends_on: foo
  script: echo 'Bar'
```

Reply:
Thanks, that's a pretty good catch!

https://github.com/cirruslabs/cirrus-cli/pull/21/commits/bf7f1bfe9e3f2f6ab5e0b49c3bda057979282a21

I've also thought about additionally preventing this in the `Build.taskHasUnresolvedDependencies()` itself, but I think that this will only be a burden with passing down the errors, since currently we're trusting the `.cirrus.yml` parser and in the future the parser will be completely in-house.